test/uiauto/base.js

```diff
@@ -72,40 +66,39 @@ async function newInstruments (bootstrapFile) {
 async function init (bootstrapFile, sock) {
   let proxy = new UIAutoClient(sock);
   let instruments = await newInstruments(bootstrapFile);
-  instruments.start(null, async () => {
-    await proxy.safeShutdown();
-    throw new Error('Unexpected shutdown of instruments');
+  instruments.on('exit', () => {
+    Promise.resolve(async () => {
```

Comment:
Yes you want to do that and handle the error, the only alternative is to loose track of the error completely. So the rule of thumb for `setTimeout`, `on` and the like is to make the exception throw, and then handle it.

In this case it is an abnormal exit, I am gonna switch the Instruments so that it uses a onShutdow promise instead, making onShutdown throws on unexpected shutdown.